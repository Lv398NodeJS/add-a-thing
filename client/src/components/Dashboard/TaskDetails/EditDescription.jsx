import React from 'react';
import { Button, Container, Form } from 'react-bootstrap';
import del from '../../assets/delete.svg';

export default class EditDescription extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      editDescription: props.editDescription,
    };
  }

  handleSaveDescription = () => {
    const { closeEditDescriptionField, saveDescription } = this.props;
    const { editDescription, taskDescription } = this.state;
    saveDescription(false, taskDescription, false, false);
    closeEditDescriptionField();
    this.setState({ editDescription: !editDescription });
  };

  closeDescription = () => {
    const { closeEditDescriptionField } = this.props;
    closeEditDescriptionField();
  };

  handleSave = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value,
    });
  };

  render() {
    const { description } = this.props;
    const { taskDescription } = this.state;
    return (
      <Container>
        <Form.Control
          name="taskDescription"
          as="textarea"
          placeholder="Description"
          defaultValue={description}
          value={taskDescription}
          onChange={this.handleSave}
        />
        <Button
          className="button-save-task-details"
          onClick={this.handleSaveDescription}
        >
          {'Save description'}
        </Button>
        <Button
          className="cancel-button"
          onClick={this.closeDescription}
        >
          <img src={del} alt="close" className="delete-icon" />
        </Button>
      </Container>
    );
  }
}
