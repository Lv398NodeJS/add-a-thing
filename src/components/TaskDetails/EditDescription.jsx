import React from 'react';
import { Button, Container, Form } from 'react-bootstrap';

export default class EditDescription extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      editDescription: props.editDescription,
    };
  }

  handleSaveDescription = () => {
    const { closeEditDescriptionField, saveDescription } = this.props;
    const { editDescription } = this.state;
    saveDescription(false, this.taskDescription.value, false, false);
    closeEditDescriptionField();
    this.setState({ editDescription: !editDescription });
  };

  render() {
    const { description } = this.props;
    return (
      <Container>
        <Form.Label htmlFor="taskDescription">Description: </Form.Label>
        <Form.Control
          name="taskDescription"
          as="textarea"
          placeholder="Description"
          defaultValue={description}
          ref={(taskDescription) => {
            this.taskDescription = taskDescription;
          }}
        />
        <Button onClick={this.handleSaveDescription}>
          {'Save description'}
        </Button>
      </Container>
    );
  }
}
