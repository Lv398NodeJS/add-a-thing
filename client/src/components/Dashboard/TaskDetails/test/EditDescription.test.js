import React from 'react';
import { shallow } from 'enzyme';
import EditDescription from '../EditDescription';

describe('EditDescription', () => {
  it('should render correctly', () => {
    const component = shallow(<EditDescription />);
    expect(component).toMatchSnapshot();
  });
});
