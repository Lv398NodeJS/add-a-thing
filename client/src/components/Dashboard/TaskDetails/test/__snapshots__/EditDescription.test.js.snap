// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditDescription should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditDescription />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          as="textarea"
          name="taskDescription"
          onChange={[Function]}
          placeholder="Description"
        />,
        <ForwardRef
          active={false}
          className="button-save-task-details"
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          Save description
        </ForwardRef>,
        <ForwardRef
          active={false}
          className="cancel-button"
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          <img
            alt="close"
            className="delete-icon"
            src="delete.svg"
          />
        </ForwardRef>,
      ],
      "fluid": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "textarea",
          "defaultValue": undefined,
          "name": "taskDescription",
          "onChange": [Function],
          "placeholder": "Description",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Feedback": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "type": "valid",
            },
            "displayName": "Feedback",
            "propTypes": Object {
              "as": [Function],
              "type": [Function],
            },
            "render": [Function],
          },
          "displayName": "FormControl",
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": false,
          "children": "Save description",
          "className": "button-save-task-details",
          "disabled": false,
          "onClick": [Function],
          "type": "button",
          "variant": "primary",
        },
        "ref": null,
        "rendered": "Save description",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "active": false,
            "disabled": false,
            "type": "button",
            "variant": "primary",
          },
          "displayName": "Button",
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": false,
          "children": <img
            alt="close"
            className="delete-icon"
            src="delete.svg"
          />,
          "className": "cancel-button",
          "disabled": false,
          "onClick": [Function],
          "type": "button",
          "variant": "primary",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "close",
            "className": "delete-icon",
            "src": "delete.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "active": false,
            "disabled": false,
            "type": "button",
            "variant": "primary",
          },
          "displayName": "Button",
          "render": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "fluid": false,
      },
      "displayName": "Container",
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            as="textarea"
            name="taskDescription"
            onChange={[Function]}
            placeholder="Description"
          />,
          <ForwardRef
            active={false}
            className="button-save-task-details"
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            Save description
          </ForwardRef>,
          <ForwardRef
            active={false}
            className="cancel-button"
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            <img
              alt="close"
              className="delete-icon"
              src="delete.svg"
            />
          </ForwardRef>,
        ],
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "textarea",
            "defaultValue": undefined,
            "name": "taskDescription",
            "onChange": [Function],
            "placeholder": "Description",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Feedback": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "type": "valid",
              },
              "displayName": "Feedback",
              "propTypes": Object {
                "as": [Function],
                "type": [Function],
              },
              "render": [Function],
            },
            "displayName": "FormControl",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": false,
            "children": "Save description",
            "className": "button-save-task-details",
            "disabled": false,
            "onClick": [Function],
            "type": "button",
            "variant": "primary",
          },
          "ref": null,
          "rendered": "Save description",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "active": false,
              "disabled": false,
              "type": "button",
              "variant": "primary",
            },
            "displayName": "Button",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": false,
            "children": <img
              alt="close"
              className="delete-icon"
              src="delete.svg"
            />,
            "className": "cancel-button",
            "disabled": false,
            "onClick": [Function],
            "type": "button",
            "variant": "primary",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "close",
              "className": "delete-icon",
              "src": "delete.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "active": false,
              "disabled": false,
              "type": "button",
              "variant": "primary",
            },
            "displayName": "Button",
            "render": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "fluid": false,
        },
        "displayName": "Container",
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
